edx_exams/apps/core/api.py

```diff
@@ -26,6 +33,22 @@ def get_attempt_by_id(attempt_id):
 
     return None
 
+def get_attempt_by_user_id(user_id):
+    """ 
+    Function to fetch a user's latest exam attempt
+    that has a status of 'started' or 'ready_to_submit'.
+    """
+
+    user_attempts = ExamAttempt.get_attempts_by_user(user_id)
+    
+    in_progress_attempts = user_attempts.filter(status__in=[ExamAttemptStatus.started, ExamAttemptStatus.ready_to_submit])
+    
+    if in_progress_attempts.count() > 0:
+        latest_in_progress_attempt = in_progress_attempts.order_by('-start_time')[0]
+        serialized_attempt = ExamAttemptSerializer(latest_in_progress_attempt)
```

Comment:
Serializer's shouldn't be imported into the core api - I modified this in https://github.com/edx/edx-exams/pull/78.

Reply:
Serializer moved to views.py